src/fenics_constitutive/solver.py

```diff
@@ -282,13 +301,33 @@ def form(self, x: PETSc.Vec) -> None:
                 self.submesh_maps[k].map_to_parent(self._stress[k], self.stress_1)
                 self.submesh_maps[k].map_to_parent(self._tangent[k], self.tangent)
 
+        if self.mesh_update:
+            self._u.function_space.mesh.geometry.x[:] -= midpoint_displacement.reshape(-1, 3)
+
         self.stress_1.x.scatter_forward()
         self.tangent.x.scatter_forward()
 
     def update(self) -> None:
         """
         Update the current displacement, stress and history.
         """
+        for k, (law, cells) in enumerate(self.laws):
+            if self.co_rotation:
+                self.stress_rotate(del_grad_u=self._del_grad_u[k].x.array, mandel_stress=self.stress_1.x.array)
+
+        if self.mesh_update:
+
+            # Update to current configuration
```

Comment:
this code is nearly identically with the one in self.build_history (factor 0.5). Would it make sense to put that in an separate  class function? And do we need to do the interpolation twice?